public/app/plugins/panel/annolist/AnnoListPanel.tsx

```diff
@@ -74,7 +75,7 @@ export class AnnoListPanel extends PureComponent<Props, State> {
 
     let timeInfo = '';
     if (options.onlyInTimeRange) {
-      const { timeRange } = this.props;
+      const timeRange = getTimeSrv().timeRange();
```

Comment:
Time range is passed via props, it should be up to date, if not we need to fix it in PanelChrome

Reply:
So, not sure my understanding of this is correct, but what's happening as far as I can tell is this:
In the `renderPanel` method of `PanelChrome`, we pass the following as the `timeRange` prop to `PanelComponent`:
```ts
const timeRange = data.timeRange || this.timeSrv.timeRange();
```
The problem seems to be that `data` is only updated when a query is run (via `onDataUpdate`) which doesn't happen with the AnnoListPanel since it's not for visualizing datasource query results. Instead, `data.timeRange` is always just what it is initialized to, `getDefaultTimeRange()`.
I think if we instead initialized it to `undefined` that would fix the underlying problem, but let me know if I've made a mistake in my reasoning.

Reply:
Yes, the fix needs to be done in PanelChrome so this is fixed for non data panels

Reply:
Some thoughts:
Typically, with plugins that make queries, the `timeRange` is included as part of the resulting `PanelData`, and this new `timeRange` is passed to the `PanelComponent` when `PanelChrome` renders the panel.
For other plugins, like the `AnnoList` panel, what is currently happening is `timeRange` just stays as the initialized value (`getDefaultTimeRange()`), or if you change the panel to a querying plugin and modify the time range then switch back to the `AnnoList`, it will now have that `timeRange`. That latter point highlights a problem with just initializing `timeRange` to undefined, besides typing considerations, as we just end up stuck with the previous plugin’s time range if we switch from it to `AnnoList`.
The only solution I can think of (that's not potentially a breaking change) besides what's already been committed to this PR is for the plugin to make a "dummy" query but I imagine this would be very hacky and not ideal.

Reply:
So onRefresh when we do not execute the query: https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/dashgrid/PanelChrome.tsx#L212 

We should update the time range (from timeSrv) before rendering the panel